scripts/ci/check_compliance.py

```diff
@@ -826,14 +830,23 @@ def run(self):
         if not py_files:
             return
 
-        pylintcmd = ["pylint", "--rcfile=" + pylintrc] + py_files
+        python_environment = os.environ.copy()
+        if "PYTHONPATH" in python_environment:
+            python_environment["PYTHONPATH"] = check_script_dir + ":" + \
+                                               python_environment["PYTHONPATH"]
+        else:
+            python_environment["PYTHONPATH"] = check_script_dir
```

Comment:
You could always set and only handle the extend case:

```
python_environment["PYTHONPATH"] = check_script_dir 
if "PYTHONPATH" in os.environ:
  python_environment["PYTHONPATH"] += ":" + os.environ['PYTHONPATH"]
```

saves an "else", up to you really.